File: src/main/java/AbstractFileSystemNode.java
public abstract class AbstractFileSystemNode implements FileSystemNode{
    protected FileSystemNode parent = null;
    protected String name;

    public final FileSystemNode getParent(){
        return parent;
    }

    public String getName(){
        return name;
    }

    public final String getPath(){
        if (getParent() != null)
            return getParent().getPath() + "/" + name;
        return name;
    }
}

File: src/main/java/AbstractFileSystemNode.kt
abstract class AbstractFileSystemNode : FileSystemNode {
    protected var parent_value: FileSystemNode? = null
    protected var name_value: String? = null
    override fun getParent(): FileSystemNode? {
        return parent_value
    }

    fun setParent(parent: FileSystemNode ) {
        parent_value = parent
    }

    override fun getName(): String {
        return name_value!!
    }

    override fun getPath(): String {
        return if (getParent() != null) getParent()?.getPath() + "/" + getName() else getName()!!
    }
}